### PR TITLE
make default test parameters unique

### LINKED ISSUE #154: Enable simultaneous test execution
There are many test classes that deploy resources that requires unique name or some other unique attribute. This slows down development - if two developers are executing tests localy, it becomes a problem. We need to have a unique string attached to resource names.

### PR DESCRIPTION
**Pull request by laDok8**
Closes #154 
make default test parameters for azure-group and injected instances unique via maven timestamp

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -115,6 +122,8 @@
                         <failIfNoTests>false</failIfNoTests>
                         <systemPropertyVariables>
                             <java.io.tmpdir>${project.build.directory}</java.io.tmpdir>
+                            <test.instance.name>${test.instance.name}</test.instance.name>
+                            <azure.group>${azure.group}</azure.group>
```
Comment (istraka) [inline pom.xml:126]: IMO it is not appropriate to define module specific property in root pom. Can you move the logic into module's poms? Also timestamp properties.

File: aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java [hunk 2]
```diff
@@ -5,7 +5,7 @@ public class AwsTestConstants {
     public static final String TAG = "sunstoneKeyTag";
     public static final String NAME_1 = "sunstoneKeyName1";
     public static final String NAME_2 = "sunstoneKeyName2";
-    public static final String instanceName = "${non.existing:eapSunstoneInjectInstance}";
+    public static final String instanceName = "${test.instance.name:eapSunstoneInjectInstance}";
```
Comment (istraka) [inline aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java:8]: It may not be clear at the first sight where does `test.instance.name` comes from. I would expect `sunstone.properties`. Can you define it in there? Something like test.instance.name=maven.pom.whatever? This may not be ideal naming, feel free to come up with something more appropriate.

**Review by istraka — CHANGES_REQUESTED** [review]
Looks good. Just two minor issues.

File: aws-wildfly/pom.xml [hunk 3]
```diff
@@ -13,11 +13,25 @@
...
+
     <build>
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <mvn.pom.test.instance.name>${test.instance.name}</mvn.pom.test.instance.name>
+                        <azure.group>${azure.group}</azure.group>
```
Comment (istraka) [inline aws-wildfly/pom.xml:32]: azure.group in aws module

File: aws-wildfly/pom.xml [hunk 4]
```diff
@@ -13,11 +13,25 @@
 
     <name>${project.artifactId}</name>
 
+    <properties>
+        <!-- to guarantee test instance Name uniqueness, supports: https://github.com/wildfly-extras/sunstone/issues/154 -->
+        <timestamp>${maven.build.timestamp}</timestamp>
+        <maven.build.timestamp.format>yyyyMMddHHmmss</maven.build.timestamp.format>
+        <test.instance.name>eapSunstoneInjectInstance-${timestamp}</test.instance.name>
```
Comment (istraka) [inline aws-wildfly/pom.xml:20]: Can you set (form) the instance name in sunstone.properties? and propagate timestamp into it? I would also consider renaming `timestamp` to `testrun`

**Review by istraka — CHANGES_REQUESTED** [review]
Thanks, looks good. I would just rearange properties.

**Review by istraka — CHANGES_REQUESTED** [review]
Thanks, but we need to have azure test run in separate groups. If you end up with fields with same annotation, feel free to remove them.

File: azure/src/test/java/sunstone/azure/armTemplates/MultipleAzArmTemplatesRepetableTest.java [hunk 5]
```diff
@@ -6,26 +6,30 @@
...
                 @Parameter(k = "vnetTag", v = AzureTestConstants.VNET_TAG)
         },
-                template = "sunstone/azure/armTemplates/vnet.json", region = "eastus2", group = GROUP)
+                template = "sunstone/azure/armTemplates/vnet.json", region = "eastus2", group = GROUP_PREFIX + deployGroup)
 })
 public class MultipleAzArmTemplatesRepetableTest {
-    static final String GROUP = "MultipleAzArmTemplatesRepetableTest";
+    static final String GROUP_PREFIX = "MultipleAzArmTemplatesRepetableTest-";
+
+    @SunstoneProperty(expression = GROUP_PREFIX + deployGroup)
```
[the remark this answers is not in the document]
Comment (laDok8) [reply azure/src/test/java/sunstone/azure/armTemplates/MultipleAzArmTemplatesRepetableTest.java:31]: what if we wanted to run tests in parallel? Adding (this) prefix to each class would be the only option. Or am I overthinking it and this parallelism is unnecessary?

File: aws/src/test/java/aws/cloudformation/AwsTestConstants.java [hunk 6]
```diff
@@ -5,7 +5,7 @@ public class AwsTestConstants {
     public static final String TAG = "sunstoneKeyTag";
     public static final String NAME_1 = "sunstoneKeyName1";
```
Comment (istraka) [inline aws/src/test/java/aws/cloudformation/AwsTestConstants.java:6]: This also need to be run specific, like `instanceName`

File: aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java [hunk 7]
```diff
@@ -5,7 +5,7 @@ public class AwsTestConstants {
     public static final String TAG = "sunstoneKeyTag";
```
Comment (istraka) [inline aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java:5]: These 3 are no longer used, can you remove them?

File: azure/src/test/java/sunstone/azure/armTemplates/di/AzVmTests.java [hunk 8]
```diff
@@ -11,40 +11,37 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
-import static sunstone.azure.armTemplates.di.AzVmTests.group;
 
 
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eap.json", group = group, perSuite = true)
+        template = "sunstone/azure/armTemplates/eap.json", perSuite = true)
```
Comment (istraka) [inline azure/src/test/java/sunstone/azure/armTemplates/di/AzVmTests.java:20]: Set the `perSuite` to `false`, please. Test class is not urin within a suite.

File: aws/src/test/java/aws/cloudformation/AwsTestConstants.java (hunk 6 above)
Comment (istraka) [inline aws/src/test/java/aws/cloudformation/AwsTestConstants.java:6]: These two also need to set like instanceName

File: aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java (hunk 7 above)
Comment (istraka) [inline aws-wildfly/src/test/java/aws/cloudformation/AwsTestConstants.java:5]: These 3 are no longer used, can you remove them?

File: azure/src/test/java/sunstone/azure/armTemplates/di/AzPgSqlTests.java [hunk 9]
```diff
@@ -10,45 +10,41 @@
...
-import static sunstone.azure.armTemplates.di.AzVmTests.group;
 
 
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "username", v = AzPgSqlTests.PGSQL_USER),
         @Parameter(k = "password", v = AzPgSqlTests.PGSQL_PASSWORD),
         @Parameter(k = "serverName", v = AzPgSqlTests.PGSQL_NAME),
 },
-        template = "sunstone/azure/armTemplates/posgresql.json", group = group, perSuite = true)
+        template = "sunstone/azure/armTemplates/posgresql.json", perSuite = true)
```
Comment (istraka) [inline azure/src/test/java/sunstone/azure/armTemplates/di/AzPgSqlTests.java:20]: Can you set `perSuite` to `false`?

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/WebAppDeploySuiteTests.java [hunk 10]
```diff
@@ -14,5 +14,4 @@
 @Suite
 @SelectClasses({AzureWebAppDeployFirstTest.class, AzureWebAppUndeployedSecondTest.class})
 public class WebAppDeploySuiteTests {
-    public static final String webAppDeployGroup = "deploytestWebApp";
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/WebAppDeploySuiteTests.java:17]: Tests run within this suite must use same resource group. Please revert this change.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/VmDomainDeploySuiteTests.java [hunk 11]
```diff
@@ -13,5 +13,4 @@
 @Suite
 @SelectClasses({AzureDomainVmDeployFirstTest.class, AzureDomainVmUndeployedSecondTest.class})
 public class VmDomainDeploySuiteTests {
-    public static final String vmDeployGroup = "deploytestVM";
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/VmDomainDeploySuiteTests.java:16]: Tests run within this suite must use same resource. Please revert this change and use this group in AzureDomainVmDeployFirstTest and AzureDomainVmUndeployedSecondTest.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/suitetests/AzureVmUndeployedSecondTest.java [hunk 12]
```diff
@@ -22,9 +22,10 @@
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eap.json", group = VmDeploySuiteTests.vmDeployGroup, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/suitetests/AzureVmUndeployedSecondTest.java:25]: This needs to be reverted.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/suitetests/AzureVmDeployFirstTest.java [hunk 13]
```diff
@@ -18,23 +18,22 @@
 
 import java.io.IOException;
 
-import static sunstone.azure.armTemplates.archiveDeploy.vm.VmDeploySuiteTests.vmDeployGroup;
-
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eap.json", group = vmDeployGroup, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/suitetests/AzureVmDeployFirstTest.java:27]: this needs to be reverted

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/suitetests/AzureDomainVmUndeployedSecondTest.java [hunk 14]
```diff
@@ -22,9 +23,10 @@
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eapDomain.json", group = VmDomainDeploySuiteTests.vmDeployGroup, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/suitetests/AzureDomainVmUndeployedSecondTest.java:25]: This needs to be reverted.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/suitetests/AzureDomainVmDeployFirstTest.java [hunk 15]
```diff
@@ -18,24 +18,24 @@
 import sunstone.azure.annotation.AzureVirtualMachine;
 import sunstone.azure.annotation.WithAzureArmTemplate;
 import sunstone.azure.armTemplates.AzureTestConstants;
-import static sunstone.azure.armTemplates.archiveDeploy.vm.VmDeploySuiteTests.vmDeployGroup;
 
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eapDomain.json", group = vmDeployGroup, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vmDomain/suitetests/AzureDomainVmDeployFirstTest.java:27]: This needs to be reverted.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/VmDeploySuiteTests.java [hunk 16]
```diff
@@ -13,5 +13,4 @@
 @Suite
 @SelectClasses({AzureVmDeployFirstTest.class, AzureVmUndeployedSecondTest.class})
 public class VmDeploySuiteTests {
-    public static final String vmDeployGroup = "deploytestVM";
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/vm/VmDeploySuiteTests.java:16]: Revert this change, please.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/suitetests/AzureWebAppDeployFirstTest.java [hunk 17]
```diff
@@ -20,16 +19,17 @@
 import java.io.IOException;
 
 @WithAzureArmTemplate(template = "sunstone/azure/armTemplates/eapWebApp.json",
-        parameters = {@Parameter(k = "appName", v = AzureTestConstants.instanceName)}, group = WebAppDeploySuiteTests.webAppDeployGroup, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/suitetests/AzureWebAppDeployFirstTest.java:23]: Use group defined in the suite class.

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/suitetests/AzureWebAppUndeployedSecondTest.java [hunk 18]
```diff
@@ -14,15 +13,17 @@
...
 
 import java.io.IOException;
+import static sunstone.azure.armTemplates.AzureTestConstants.deployGroup;
 
 /**
  * The test is supposed to run after AzureWebAppDeployFirstTest and verifies undeploy operation
  */
 @WithAzureArmTemplate(template = "sunstone/azure/armTemplates/eapWebApp.json",
-        parameters = {@Parameter(k = "appName", v = AzureTestConstants.instanceName)}, group = WebAppDeploySuiteTests.webAppDeployGroup, perSuite = true)
+        parameters = {@Parameter(k = "appName", v = AzureTestConstants.instanceName)}, group = AzureWebAppUndeployedSecondTest.groupName, perSuite = true)
```
Comment (istraka) [inline azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/suitetests/AzureWebAppUndeployedSecondTest.java:22]: Define group in the suite class, please.

**Review by istraka — CHANGES_REQUESTED** [review]

File: azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/WebAppDeploySuiteTests.java (hunk 10 above)
Comment (laDok8) [reply azure-wildfly/src/test/java/sunstone/azure/armTemplates/archiveDeploy/webapp/WebAppDeploySuiteTests.java:17]: I reverted to suite common name, but kept it unique, feel free to close this. I'll mark duplicates of this as resolved

File: azure/src/test/java/sunstone/azure/armTemplates/di/AzureWebAppTests.java [hunk 19]
```diff
@@ -21,27 +21,25 @@
 import static org.assertj.core.api.Assertions.assertThat;
 
 @WithAzureArmTemplate(template = "sunstone/azure/armTemplates/eapWebApp.json",
-        parameters = {@Parameter(k = "appName", v = AzureTestConstants.instanceName)}, group = AzureWebAppTests.group)
```
Comment (istraka) [inline azure/src/test/java/sunstone/azure/armTemplates/di/AzureWebAppTests.java:24]: Can we have unique test class specific group? Like in `MultipleAzArmTemplatesRepetableTest`.

File: azure/src/test/java/sunstone/azure/armTemplates/di/AzVmTests.java [hunk 20]
```diff
@@ -11,40 +11,37 @@
...
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static sunstone.azure.armTemplates.di.AzVmTests.group;
 
 
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "virtualMachineName", v = AzureTestConstants.instanceName),
         @Parameter(k = "imageRefId", v = AzureTestConstants.IMAGE_REF)
 },
-        template = "sunstone/azure/armTemplates/eap.json", group = group, perSuite = true)
```
Comment (istraka) [inline azure/src/test/java/sunstone/azure/armTemplates/di/AzVmTests.java:21]: Can we have unique test class specific group? Like in `MultipleAzArmTemplatesRepetableTest`.

File: azure/src/test/java/sunstone/azure/armTemplates/di/AzPgSqlTests.java [hunk 21]
```diff
@@ -10,45 +10,41 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
-import static sunstone.azure.armTemplates.di.AzVmTests.group;
 
 
 @WithAzureArmTemplate(parameters = {
         @Parameter(k = "username", v = AzPgSqlTests.PGSQL_USER),
         @Parameter(k = "password", v = AzPgSqlTests.PGSQL_PASSWORD),
         @Parameter(k = "serverName", v = AzPgSqlTests.PGSQL_NAME),
 },
-        template = "sunstone/azure/armTemplates/posgresql.json", group = group, perSuite = true)
```
Comment (istraka) [inline azure/src/test/java/sunstone/azure/armTemplates/di/AzPgSqlTests.java:21]: Can we have unique test class specific group? Like in `MultipleAzArmTemplatesRepetableTest`.

**Review by istraka — CHANGES_REQUESTED** [review]
Last few bits...

**Review by istraka — APPROVED** [review]
